src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -1,22 +1,69 @@
 
-import collection.JavaConverters._
+import scala.collection.JavaConverters._
 
 class Neo4jQueryWriteStrategy(private val saveMode: SaveMode) extends Neo4jQueryStrategy {
   override def createStatementForQuery(options: Neo4jOptions): String =
-    s"""UNWIND ${"$"}events AS event
+    s"""UNWIND ${"$"}events AS $BATCH_VARIABLE
        |${options.query.value}
        |""".stripMargin
 
-  override def createStatementForRelationships(options: Neo4jOptions): String = throw new UnsupportedOperationException("TODO implement method")
+  override def createStatementForRelationships(options: Neo4jOptions): String = {
+    val relationshipKeyword = saveMode match {
+      case SaveMode.Overwrite => "MERGE"
+      case SaveMode.ErrorIfExists => "CREATE"
+      case _ => throw new UnsupportedOperationException(s"SaveMode $saveMode not supported")
+    }
+
+    val sourceKeyword = options.relationshipMetadata.sourceWriteMode match {
+      case NodeSaveMode.Overwrite => "MERGE"
+      case NodeSaveMode.ErrorIfExists => "CREATE"
+      case NodeSaveMode.Match => "MATCH"
+      case _ => throw new UnsupportedOperationException(s"Source SaveMode $saveMode not supported")
+    }
+
+    val targetKeyword = options.relationshipMetadata.targetWriteMode match {
+      case NodeSaveMode.Overwrite => "MERGE"
+      case NodeSaveMode.ErrorIfExists => "CREATE"
+      case NodeSaveMode.Match => "MATCH"
+      case _ => throw new UnsupportedOperationException(s"Target SaveMode $saveMode not supported")
+    }
+
+    val relationship = options.relationshipMetadata.relationshipType.quote()
+
+    val sourceLabels = options.relationshipMetadata.source.labels
+      .map(_.quote())
+      .mkString(":")
+
+    val targetLabels = options.relationshipMetadata.target.labels
+      .map(_.quote())
+      .mkString(":")
+
+    val sourceKeys = options.relationshipMetadata.source.nodeKeys.map(key => {
+      s"${key._2.quote()}:$BATCH_VARIABLE.source.${Neo4jWriteMappingStrategy.KEYS}.${key._1.removeAlias().quote()}"
+    }).mkString(", ")
+    val targetKeys = options.relationshipMetadata.target.nodeKeys.map(key => {
+      s"${key._2.quote()}:$BATCH_VARIABLE.target.${Neo4jWriteMappingStrategy.KEYS}.${key._1.removeAlias().quote()}"
+    }).mkString(", ")
+
+    val sourceQueryPart = createQueryPart(sourceKeyword, sourceLabels, sourceKeys, Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS)
+    val targetQueryPart = createQueryPart(targetKeyword, targetLabels, targetKeys, Neo4jUtil.RELATIONSHIP_TARGET_ALIAS, Seq(Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS))
+
+    s"""UNWIND ${"$"}events AS $BATCH_VARIABLE
+       |$sourceQueryPart
+       |$targetQueryPart
+       |$relationshipKeyword (${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS})-[${Neo4jUtil.RELATIONSHIP_ALIAS}:${relationship}]->(${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS})
+       |SET ${Neo4jUtil.RELATIONSHIP_ALIAS} = $BATCH_VARIABLE.${Neo4jUtil.RELATIONSHIP_ALIAS}.${Neo4jWriteMappingStrategy.PROPERTIES}
```

Comment:
io ricordo c'era il `+`, comunque (ri)mettilo
```suggestion
       |SET ${Neo4jUtil.RELATIONSHIP_ALIAS} += $BATCH_VARIABLE.${Neo4jUtil.RELATIONSHIP_ALIAS}.${Neo4jWriteMappingStrategy.PROPERTIES}
```